pvlib/irradiance.py

```diff
@@ -2915,3 +2916,68 @@ def dni(ghi, dhi, zenith, clearsky_dni=None, clearsky_tolerance=1.1,
             (zenith < zenith_threshold_for_zero_dni) &
             (dni > max_dni)] = max_dni
     return dni
+
+
+def component_sum_irradiance(solar_zenith,
+                             ghi=None,
+                             dhi=None,
+                             dni=None,
+                             dni_clear=None):
+    r"""
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the other associated series passed are used to
+    calculate the missing series value.
+
+    The "component sum" or "closure" equation relates the three
+    primary irradiance components as follows:
+
+    .. math::
+
+       GHI = DHI + DNI \cos(\theta_z)
+
+    Parameters
+    ----------
+    solar_zenith : Series
+        Refraction-corrected zenith angles in decimal
+        degrees, with datetime index. Angles must be >=0 and <=180. Must have
+        the same datetime index as ghi, dhi, and dni series, when available.
+    ghi : Series, (optional, default None)
```

Comment:
And likewise for the others (see #1574)
```suggestion
    ghi : Series, optional
```

Reply:
Updated in https://github.com/pvlib/pvlib-python/pull/1567/commits/e595465e6f678866a3bde3fa070f9b14d2a5347f